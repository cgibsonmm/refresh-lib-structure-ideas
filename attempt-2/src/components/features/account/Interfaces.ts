import { Nullable } from '@/utils/helperTypes';
export interface AccountResponse {
  account: Account;
  meta: Meta;
}

interface Meta {
  status: number;
  free_user: boolean;
  mobile_free: boolean;
  subscription_state: string;
  notification?: SiteNotification | null;
}
interface SiteNotification {
  message: string;
  title: string;
}

export interface Account {
  active_ab_tests: ActiveAbTests;
  alert_me: AlertMe;
  billing_info: BillingInfo;
  legal_doc_info: LegalDocInfo;
  notification?: AccountNotification | null;
  report_tags: ReportTags;
  retention_info: RetentionInfo;
  sandboxed: boolean;
  search_counts: SearchCounts;
  staff_info: StaffInfo;
  subscription_info: SubscriptionInfo;
  upgrade_info: UpgradeInfo;
  user_info: UserInfo;
  user_settings: UserSettings;
}

export interface UserInfo {
  first_name: string | null;
  last_name: string | null;
  email: string;
  phone_number: string;
  optout_options: boolean;
  user_code: string;
  optout_deleted: boolean;
  join_date: string;
  signup_type: string;
  design_name: string;
  design_version: string;
  tos_version_bad: boolean;
  pp_version_bad: boolean;
  noncompliant: boolean;
  tos_pending: boolean;
  'shut_down?': boolean;
  hide_person_video: boolean;
  hide_property_video: boolean;
  friends_and_family: boolean;
  user_in_recycling: boolean;
  recycling_date: string | null;
  is_self_searcher: boolean;
  seats: Seat[];
  seat_status: string | null;
  invite_count: number;
  has_connected_vehicles: boolean;
}

export interface SubscriptionInfo {
  free_user: boolean;
  'mobile_free?': boolean;
  public_plan_type: string;
  subscription_plan_unique_name: string;
  subscription_plan_internal_name: string;
  subscription_state: string;
  in_billable_state: boolean;
  recurring: boolean;
  public_price_description: string;
  normalize_date: string;
  monitor_limit: number;
  remaining_monitors: number;
  credits_remaining: number;
  monthly_reports_remaining: number;
  monthly_report_limit: number;
  recycling_reports_remaining: number;
  subscription_features: SubscriptionFeatures;
  money_amount: string;
  'show_unlimited_reports?': boolean;
  renewal_period: number;
  renewal_period_type: string;
}

export type SubscriptionFeatures = Nullable<{
  monthly_report_limit: number;
  consumer_report_block_plan_name: string;
  unlimited_reports: boolean;
  unlimited_monitors: boolean;
  report_tier: string;
  monitor_limit: number;
  daily_report_soft_limit: number;
  monthly_bjl_limit: number;
  comp_reports_enabled: boolean;
  alertme_enabled: boolean;
  unlimited_bankruptcies: boolean;
  unlimited_liens_and_judgments: boolean;
  lien_and_judgment_limit: number;
  six_month_transaction_multiplier: number;
  default_design: string;
  default_search_bar: string;
  modal_family: string;
  bankruptcy_limit: number;
  session_lock_enabled: boolean;
  pdf_access: boolean;
  pro_account: boolean;
  hourly_report_soft_limit: number;
  active_reports_limit: number;
  api_access: boolean;
}>;

export interface SubscriptionPlan {
  amazon_id: string | null;
  amount: number;
  app_only: string;
  apple_id: string | null;
  google_id: string | null;
  name: string;
  title: string;
  subtitle: string;
  recurring: boolean;
  renewal_period: number;
  renewal_period_type: string;
  report_tier: string;
  unique_key: string;
  internal_name: string;
  discount: string | null;
  public_price_description: string;
  default: boolean;
  monthly_report_limit: number;
}

export interface UpgradeInfo {
  upgrade_credits: number;
}

export interface SearchCounts {
  replenishment_date: string;
  allocated_bankruptcies: number;
  allocated_liens_and_judgments: number;
  remaining_bankruptcies: number;
  remaining_liens_and_judgments: number;
}

export interface LegalDocInfo {
  active_tos_version: number;
  active_privacy_policy_version: number;
}

export interface BillingInfo {
  card_type?: string;
  name?: string;
  card_number?: string;
  street?: string;
  city?: string;
  state?: string;
  postal_code?: string;
  country?: string;
}

export interface AlertMe {
  has_access: boolean;
  settings: AlertMeSettings;
}

export interface AlertMeSettings {
  id: number;
  digest_enabled: boolean;
  push_digest_enabled: boolean;
  instant_enabled: boolean;
  push_instant_enabled: boolean;
  next_digest_mailing_at: string;
  created_at: string;
  updated_at: string;
}

export type RetentionInfo = Nullable<{
  ran_report_on_self: boolean;
  ran_report_on_multiple_days: boolean;
  ran_report_on_family_member: boolean;
  ran_reverse_phone_search: boolean;
  ran_email_search: boolean;
  ran_professional_contact_search: boolean;
  ran_people_search: boolean;
  ran_property_search: boolean;
  ran_estate_value_search: boolean;
  created_monitor: boolean;
  created_monitor_on_self: boolean;
  created_monitor_on_family_member: boolean;
  created_monitor_on_property: boolean;
  created_monitor_on_email: boolean;
  created_monitor_on_their_email: boolean;
  has_used_mobile_app: boolean;
  has_used_website: boolean;
  downsell_plan_offered: boolean;
  downsell_plan_accepted: boolean;
  claimed_person_report: boolean;
  ran_username_search: boolean;
  ran_neighborhood_watch_search: boolean;
  ran_vehicle_search: boolean;
  claimed_report: boolean;
}>;

export type ReportTags = Nullable<{
  self_person_report_id: string;
}>;

export interface ActiveAbTests {
  tests: ActiveAbTest[];
}

export interface ActiveAbTest {
  id: number;
  user_id: number;
  source: string;
  experiment_key: string;
  variation_key: string;
  created_at: string;
  updated_at: string;
  whitelist: string | null;
}
export type UserSettings = Nullable<{
  hide_reactivation_banner: boolean;
  hide_recurring_cancelled_banner: boolean;
  hide_bad_billing_banner: boolean;
  hide_credit_card_banner: boolean;
  hide_trial_welcome_banner: boolean;
  skip_creation_modal: boolean;
  hide_person_video: boolean;
  hide_property_video: boolean;
  hide_dark_web_banner: boolean;
  hide_dark_web_notifications: boolean;
  hide_unclaimed_money_banner: boolean;
  hide_unclaimed_money_ctas: boolean;
  hide_download_app_cta: boolean;
  hide_unclaimed_money_welcome_modal: boolean;
  hide_social_cleanup_cta: boolean;
  show_info_collection_screen: boolean;
  hide_how_bv_works_modal: boolean;
  hide_payment_update_required_modal: boolean;
  hide_onboarding_checklist_modal: boolean;
  show_higher_confidence_data_only: boolean;
  show_sales_tax_notice_modal: boolean;
  hide_rate_app_banner: boolean;
  hide_multi_seats_invite_modal: boolean;
  hide_criminal_records: boolean;
  hide_privacy_cta: boolean;
  hide_ps_vs_biz_modal: boolean;
  hide_own_fcra: boolean;
  hide_own_tos: boolean;
  show_section_learn_more: boolean;
  hide_cancel_downsell_modal: boolean;
  check_marketplace_step: boolean;
  dqs_completed: boolean;
  hide_passes_invite_modal: boolean;
  hide_sitejabber_widget: boolean;
  check_invites_step: boolean;
  check_extension_step: boolean;
  check_reward_1_step: boolean;
  check_reward_2_step: boolean;
  check_reward_3_step: boolean;
  hide_getting_started: boolean;
  high_confidence_toggle: boolean;
  check_list_completed: boolean;
  check_list_skipped: boolean;
  identity_hub_completed: boolean;
  tips_completed: boolean;
  first_report_modal_seen: boolean;
  openbay_callout_modal_seen: boolean;
  credit_hub_enrollment_completed: boolean;
}>;

export interface StaffInfo {
  admin: boolean;
  blacklister: boolean;
  staff: boolean;
  beta_tester: boolean;
}

interface Seat {
  id: string;
  email: string;
  first_name: string;
  last_name: string;
  status: string;
  created_at: string;
}

export interface AccountNotification {
  type: string;
  title: string;
  message: string;
  call_to_action: string;
}

//Payments
interface ReceiptItem {
  amount: number | string;
  description: string;
  id: number;
  purchaseable_type: string;
  receipt_id: number;
}

export interface Receipt {
  amount: number | string;
  created_at: string;
  id: number;
  method: string;
  items: ReceiptItem[];
}

export interface PaymentMethods {
  active: boolean;
  card_type: string;
  city: string;
  country: string;
  expiration_month: number;
  expiration_year: number;
  email: string;
  first_name: string;
  id: number;
  last_four: string;
  last_name: string;
  masked_card_number: string;
  payer_id: string;
  state: string;
  street1: string;
  payer_status: string;
  postal_code: string;
  description: string;
  braintree_type: string;
  token: string;
  payment_type: string;
}
export interface PaymentImages {
  master: string;
  visa: string;
  american_express: string;
  discover: string;
  genericCreditCard: string;
  paypal: string;
  applePay: string;
  googlePay: string;
  venmo: string;
}
