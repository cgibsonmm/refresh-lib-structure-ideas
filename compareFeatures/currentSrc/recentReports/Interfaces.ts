import { Nullable } from '@/utils/helperTypes';

export type RawReportType =
  | 'social_network_report'
  | 'detailed_person_report'
  | 'username_report'
  | 'reverse_phone_report'
  | 'sex_offender_report'
  | 'property_report'
  | 'detailed_business_report';

export interface RecentReportRawData {
  age?: number;
  address?: string;
  alert_me?: boolean | null;
  city?: string;
  created_at?: string;
  email?: string;
  first_name?: string;
  full_size_image?: string | null;
  has_new_info?: boolean;
  interaction?: Record<string, unknown>;
  license_plate?: string;
  last_name?: string;
  make?: string;
  middle_name?: string | null;
  mileage?: number;
  model?: string;
  permalink: string;
  phone_number?: string;
  rating?: unknown;
  report_tier?: string;
  report_type: RawReportType;
  report_upgraded?: boolean;
  state?: string;
  search_by?: string;
  service_data?: unknown;
  tag?: string | null;
  thumb?: unknown;
  updated_at?: string;
  upgrade_options?: unknown;
  vin?: string;
  year?: number;
  username?: string;
  network?: string;
  zip_code?: string;
  domain?: string;
  no_entities?: unknown;
  carrier?: string;
  bvids?: string[];
}
export interface RecentReportsRawData {
  meta: ReportsMeta;
  polling: Record<string, unknown>;
  reports: RecentReportRawData[];
}

export interface RecentReportsDataSerialized {
  reports: RecentReportSerialized[];
  meta: ReportsMeta;
  polling: Record<string, unknown>;
}

export interface RecentReportSerialized {
  id: string;
  reportUpgraded?: boolean;
  createdAt: string;
  createdDate: string;
  updatedAt: string;
  updatedDate: string;
  alertMe?: boolean | null;
  hasNewInfo: boolean | null;
  reportType: string;
  reportTypeFormatted: string;
  detailsCityState?: string;
  reportData: string;
  details: RecentReportRawData;
}

interface ReportsMeta {
  status: number;
  free_user: boolean;
  mobile_free: boolean;
  subscription_state: string;
  report_quantities: Nullable<ReportQuantities>;
}

interface ReportQuantities {
  contact_reports_count: number;
  detailed_business_reports_count: number;
  email_reports_count: number;
  estate_value_reports_count: number;
  filtered_reports_count: number;
  fraud_reports_count: number;
  link_to_next_page: number;
  link_to_prev_page: number;
  person_reports_count: number;
  phone_reports_count: number;
  property_reports_count: number;
  report_limit: number;
  sex_offender_reports_count: number;
  total_reports_count: number;
  username_reports_count: number;
  vehicle_reports_count: number;
}
