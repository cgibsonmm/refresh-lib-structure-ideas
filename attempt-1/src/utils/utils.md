# Utils

This folder will house global used Utils

I.E.

1. commaFormat
