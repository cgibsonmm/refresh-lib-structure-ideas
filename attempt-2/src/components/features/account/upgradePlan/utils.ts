/**
 * Calculates the cost per month for a plan
 *
 * @param cost Cost of the plan
 * @param renewalPeriod Number of months the plan is valid for
 * @returns Cost per month
 */
export function monthlyPlanCost(cost: number, renewalPeriod: number): number {
  return cost / renewalPeriod;
}

/**
 * Calculates the monthly cost per report for a plan
 *
 * @param cost Cost of the plan
 * @param renewalPeriod Number of months the plan is valid for
 * @param limit Monthly report limit of the plan
 * @returns Cost per report rounded to the nearest hundredth
 */
export function costPerReport(
  cost: number,
  renewalPeriod: number,
  limit: number
): number {
  const mthlyPlanCost = Math.ceil(monthlyPlanCost(cost, renewalPeriod));

  // Do not change, the rounding used here is important and was agreed with the product team
  //round to hundredths
  const costPerReport = Math.ceil((mthlyPlanCost / limit) * 100) / 100;

  return costPerReport;
}

/**
 * Calculates the savings based on the monthly cost per report of upgrading from a base plan to a new plan
 *
 * @param baseCost Cost of the base plan
 * @param baseRenewalPeriod Number of months the base plan is valid for
 * @param baseLimit Monthly report limit of the base plan
 * @param upgradeCost Cost of the upgrade plan
 * @param upgradeRenewalPeriod Number of months the new plan is valid for
 * @param upgradeLimit Monthly report limit of the upgrade plan
 * @returns Savings as a percentage
 */
export function savings(
  baseCost: number,
  baseRenewalPeriod: number,
  baseLimit: number,
  upgradeCost: number,
  upgradeRenewalPeriod: number,
  upgradeLimit: number
): number {
  const baseCostPerReport = costPerReport(
    baseCost,
    baseRenewalPeriod,
    baseLimit
  );
  const upgradeCostPerReport = costPerReport(
    upgradeCost,
    upgradeRenewalPeriod,
    upgradeLimit
  );

  // Do not change, the rounding used here is important and was agreed with the product team
  const savings = Math.floor(
    ((baseCostPerReport - upgradeCostPerReport) / baseCostPerReport) * 100
  );

  return savings;
}
