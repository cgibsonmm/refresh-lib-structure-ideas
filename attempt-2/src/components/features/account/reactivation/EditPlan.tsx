import { Text, Button } from '@/theme/index';
import { dollarize } from '@/utils/dollarize';
import { EditPlanCard } from './EditPlanCard';
import { ButtonContainer, PlansContainer, Right } from './components';
import { SubscriptionPlan } from '../Interfaces';

export function EditPlan({
  plans,
  selectedPlan,
  handleSelectedPlan,
  handleCancelClick,
  toggleEditPlan,
}: {
  plans: SubscriptionPlan[];
  selectedPlan: SubscriptionPlan;
  handleSelectedPlan: (
    selectedPLanInternalName: string,
    selectedPLanMonthlyReportLimit: number
  ) => void;
  toggleEditPlan: () => void;
  handleCancelClick: () => void;
}) {
  return (
    <PlansContainer>
      <Text variant="h2">Select from the plans below:</Text>
      {plans.map((plan) => (
        <EditPlanCard
          plan={plan}
          selectedPlan={selectedPlan}
          handleSelectedPlan={handleSelectedPlan}
          key={plan.title}
        />
      ))}
      <Right>
        <Text variant="h3">Due today</Text>
        <Text variant="h2">{dollarize(selectedPlan?.amount)}</Text>
        <Text variant="body1" color="text.secondary">
          Plus applicable sales tax
        </Text>
      </Right>
      <ButtonContainer>
        <Button
          variant="contained"
          sx={{ marginBottom: '8px' }}
          onClick={toggleEditPlan}
        >
          Select this plan
        </Button>
        <Button onClick={handleCancelClick}>Cancel selection</Button>
      </ButtonContainer>
    </PlansContainer>
  );
}
