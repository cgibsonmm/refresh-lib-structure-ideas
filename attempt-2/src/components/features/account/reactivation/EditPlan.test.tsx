import { render, screen, fireEvent } from '@testing-library/react';
import { dollarize } from '@/utils/dollarize';
import { EditPlan } from './EditPlan';
import { SubscriptionPlan } from '../Interfaces';

const plans = [
  {
    internalName: 'basic',
    title: 'Basic Plan',
    amount: 1000,
  },
  {
    internalName: 'pro',
    title: 'Pro Plan',
    amount: 2000,
  },
];

const selectedPlan = {
  internalName: 'pro plus',
  title: 'Pro Plan',
  amount: 200000,
};

const fnWithArg = jest.fn((title) => title);
const emptyFn = jest.fn();

describe('EditPlan', () => {
  it('renders', () => {
    const { asFragment } = render(
      <EditPlan
        plans={plans as unknown as SubscriptionPlan[]}
        selectedPlan={selectedPlan as unknown as SubscriptionPlan}
        handleSelectedPlan={fnWithArg}
        toggleEditPlan={emptyFn}
        handleCancelClick={emptyFn}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the list of plans', () => {
    render(
      <EditPlan
        plans={plans as unknown as SubscriptionPlan[]}
        selectedPlan={selectedPlan as unknown as SubscriptionPlan}
        handleSelectedPlan={fnWithArg}
        toggleEditPlan={emptyFn}
        handleCancelClick={emptyFn}
      />
    );

    const planNames = plans.map((plan) => screen.getByText(plan.title));
    expect(planNames).toHaveLength(plans.length);
  });

  it('displays the selected plan amount', () => {
    render(
      <EditPlan
        plans={plans as unknown as SubscriptionPlan[]}
        selectedPlan={selectedPlan as unknown as SubscriptionPlan}
        handleSelectedPlan={fnWithArg}
        toggleEditPlan={emptyFn}
        handleCancelClick={emptyFn}
      />
    );

    const amountText = screen.getByText(dollarize(selectedPlan.amount));
    expect(amountText).toBeInTheDocument();
  });

  it('calls toggleEditPlan when the "Select this plan" button is clicked', () => {
    const toggleEditPlan = jest.fn();
    render(
      <EditPlan
        plans={plans as unknown as SubscriptionPlan[]}
        selectedPlan={selectedPlan as unknown as SubscriptionPlan}
        handleSelectedPlan={fnWithArg}
        toggleEditPlan={toggleEditPlan}
        handleCancelClick={emptyFn}
      />
    );

    fireEvent.click(screen.getByText('Select this plan'));

    expect(toggleEditPlan).toHaveBeenCalled();
  });

  it('calls handleCancelClick when the "Cancel selection" button is clicked', () => {
    const handleCancelClick = jest.fn();
    render(
      <EditPlan
        plans={plans as unknown as SubscriptionPlan[]}
        selectedPlan={selectedPlan as unknown as SubscriptionPlan}
        handleSelectedPlan={fnWithArg}
        toggleEditPlan={emptyFn}
        handleCancelClick={handleCancelClick}
      />
    );

    fireEvent.click(screen.getByText('Cancel selection'));

    expect(handleCancelClick).toHaveBeenCalled();
  });
});
