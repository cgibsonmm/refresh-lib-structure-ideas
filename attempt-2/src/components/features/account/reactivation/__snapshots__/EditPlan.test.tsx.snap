// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPlan renders 1`] = `
<DocumentFragment>
  <div
    class="css-117jhq4"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
    >
      Select from the plans below:
    </h2>
    <div
      class="css-kksclf"
      role="button"
    >
      <span
        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary css-tque03-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <div
        class="css-1lxwves"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Basic Plan
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
        />
      </div>
      <div
        class="css-1ys4hkf"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
        >
          $10
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17hfm4p-MuiTypography-root"
        >
          Per month plus
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17hfm4p-MuiTypography-root"
        >
          applicable sales tax
        </p>
      </div>
    </div>
    <div
      class="css-kksclf"
      role="button"
    >
      <span
        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary css-tque03-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <div
        class="css-1lxwves"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Pro Plan
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
        />
      </div>
      <div
        class="css-1ys4hkf"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
        >
          $20
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17hfm4p-MuiTypography-root"
        >
          Per month plus
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-17hfm4p-MuiTypography-root"
        >
          applicable sales tax
        </p>
      </div>
    </div>
    <div
      class="css-1ys4hkf"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
      >
        Due today
      </h3>
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
      >
        $2000
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
      >
        Plus applicable sales tax
      </p>
    </div>
    <div
      class="css-dzsxei"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-13dyovk-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Select this plan
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Cancel selection
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
